peer/upstream/upstream.go

```diff
@@ -0,0 +1,350 @@
+// Copyright (c) 2014-2019 Bitmark Inc.
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package upstream
+
+import (
+	"encoding/binary"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/bitmark-inc/bitmarkd/blockdigest"
+	"github.com/bitmark-inc/bitmarkd/fault"
+	"github.com/bitmark-inc/bitmarkd/mode"
+	"github.com/bitmark-inc/bitmarkd/util"
+	"github.com/bitmark-inc/bitmarkd/zmqutil"
+	"github.com/bitmark-inc/logger"
+	zmq "github.com/pebbe/zmq4"
+)
+
+type UpstreamIntf interface {
```

Comment:
I also think you should name it `Upstream` and the basic implementation might be `UpstreamImpl`. It's more consistent.

Reply:
it should, but I don't want to change all other object already use Upstream

Reply:
No matter. It should be changed fore more consistent.

Reply:
If it should better do now and change all other objects then postpone and never do that.

Reply:
I think if it's a suggestion, commenter one should have clear confirm about comment that something should world. 

And yes, I have test, it works

Reply:
the single one module is inconsistent to all other existing convention, so I think it should be a question about should all others also changed? Or should we refactor to some kind of convention whenever modifying legacy code